doc/connectivity/networking/dsa.rst

```diff
@@ -0,0 +1,125 @@
+.. _dsa:
+
+Distributed Switch Architecture (DSA)
+#####################################
+
+.. contents::
+    :local:
+    :depth: 2
+
+Distributed Switch Architecture (DSA) is not something new. It has been a mature
+subsystem in Linux for many years. This document just skips the background,
+terms and any knowledge related description, as user may find all of these in
+`Linux DSA documentation`_.
+
+
+Architecture
+************
+
+Same with Linux, from network device perspective the DSA architecture is as below.
+
+.. code-block:: console
+
+                Unaware application
+              opens and binds socket
+                       |  ^
+                       |  |
+           +-----------v--|--------------------+
+           |+------+ +------+ +------+ +------+|
+           || swp0 | | swp1 | | swp2 | | swp3 ||
+           |+------+-+------+-+------+-+------+|
+           |          DSA switch driver        |
+           +-----------------------------------+
+                         |        ^
+            Tag added by |        | Tag consumed by
+           switch driver |        | switch driver
+                         v        |
+           +-----------------------------------+
+           | Unmodified host interface driver  | Software
+   --------+-----------------------------------+------------
+           |       Host interface (eth0)       | Hardware
+           +-----------------------------------+
+                         |        ^
+         Tag consumed by |        | Tag added by
+         switch hardware |        | switch hardware
+                         v        |
+           +-----------------------------------+
+           |               Switch              |
+           |+------+ +------+ +------+ +------+|
+           || swp0 | | swp1 | | swp2 | | swp3 ||
+           ++------+-+------+-+------+-+------++
+
+Host interface
+**************
+
+Host interface network devices use regular and unmodified ethernet driver working
+as DSA master port, which manages the switch through processor.
```

Comment:
Similar to linux we should phase out the term "master" and prefer "conduit", similar instead of "slave" use "user".

From https://docs.kernel.org/networking/dsa/dsa.html

> NB: for the past 15 years, the DSA subsystem had been making use of the terms “master” (rather than “conduit”) and “slave” (rather than “user”). These terms have been removed from the DSA codebase and phased out of the uAPI.

Reply:
Right. Will convert to new terms.
Thanks.